src/Cryptol/Eval/Value.hs

```diff
@@ -369,15 +397,55 @@ ilam sym f =
                      Nat i -> f i
                      Inf   -> panic "ilam" [ "Unexpected `inf`" ])
 
+-- | A finite sequence of non-VBit values. Used in 'finSeq' to
+--   safely construct a 'VSeq'.
+newtype FinSeq sym = FinSeq { fromFinSeq :: [GenValue sym] }
+  deriving Generic
```

Comment:
Please remove this `Generic` instance, as this leaks the implementation details of `FinSeq`. (Since `FinSeq` is exported abstractly without its newtype constructor, I don't think this was intended.)